src/main/java/com/powsybl/forcelayout/ForceLayout.java

```diff
@@ -159,18 +170,58 @@ public void execute() {
         LOGGER.info("Elapsed time: {}", elapsedTime / 1e9);
     }
 
-    private void applyCoulombsLaw() {
+    private Vector coulombsForce(Vector p1, Vector p2, double repulsion) {
+        Vector distance = p1.subtract(p2);
+        Vector direction = distance.normalize();
+        return direction.multiply(repulsion).divide(distance.magnitudeSquare() * 0.5 + 0.1);
+    }
+
+    private void applyCoulombsLawToPoints() {
         for (Point point : points.values()) {
+            Vector p = point.getPosition();
             for (Point otherPoint : points.values()) {
                 if (!point.equals(otherPoint)) {
-                    Vector distance = point.getPosition().subtract(otherPoint.getPosition());
-                    Vector direction = distance.normalize();
+                    point.applyForce(coulombsForce(p, otherPoint.getPosition(), repulsion));
+                }
+            }
+        }
+    }
 
-                    Vector force = direction.multiply(repulsion).divide(distance.magnitudeSquare() * 0.5 + 0.1);
+    private void applyCoulombsLawToSprings() {
+        for (Point point : points.values()) {
+            Vector p = point.getPosition();
+            for (Spring spring : springs) {
+                Point n1 = spring.getNode1();
+                Point n2 = spring.getNode2();
+                if (!n1.equals(point) && !n2.equals(point)) {
+                    Vector q1 = spring.getNode1().getPosition();
+                    Vector q2 = spring.getNode2().getPosition();
+                    Vector center = q1.add(q2.subtract(q1).multiply(0.5));
+                    Vector force = coulombsForce(p, center, repulsion * springRepulsionFactor);
```

Comment:
We could later apply this force only if the point is on (or nearly on) the spring, to avoid loosing the prior compact size

Reply:
Good idea. We can explore this in a separate pull request.